contrib/rust/src/python/pants/contrib/rust/tasks/cargo_task.py

```diff
@@ -0,0 +1,139 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import os
+
+from pants.task.task import Task
+from pants.util.process_handler import subprocess
+
+from pants.contrib.rust.targets.cargo_base_binary import CargoBaseBinary
+from pants.contrib.rust.targets.cargo_base_custom_build import CargoBaseCustomBuild
+from pants.contrib.rust.targets.cargo_base_library import CargoBaseLibrary
+from pants.contrib.rust.targets.cargo_base_proc_macro import CargoBaseProcMacro
+from pants.contrib.rust.targets.cargo_base_target import CargoBaseTarget
+from pants.contrib.rust.targets.cargo_binary import CargoBinary
+from pants.contrib.rust.targets.cargo_library import CargoLibrary
+from pants.contrib.rust.targets.cargo_synthetic_binary import CargoSyntheticBinary
+from pants.contrib.rust.targets.cargo_synthetic_custom_build import CargoSyntheticCustomBuild
+from pants.contrib.rust.targets.cargo_synthetic_library import CargoSyntheticLibrary
+from pants.contrib.rust.targets.cargo_synthetic_proc_macro import CargoSyntheticProcMacro
+from pants.contrib.rust.targets.cargo_target import CargoTarget
+from pants.contrib.rust.targets.cargo_test import CargoTest
+from pants.contrib.rust.targets.cargo_workspace import CargoWorkspace
+
+
+class CargoTask(Task):
+
+  @staticmethod
+  def manifest_name():
+    return 'Cargo.toml'
+
+  @staticmethod
+  def is_cargo_binary(target):
+    return isinstance(target, CargoBinary)
+
+  @staticmethod
+  def is_workspace(target):
+    return isinstance(target, CargoWorkspace)
+
+  @staticmethod
+  def is_cargo_workspace(target):
+    return isinstance(target, CargoWorkspace)
+
+  @staticmethod
+  def is_cargo_base_target(target):
+    return isinstance(target, CargoBaseTarget)
+
+  @staticmethod
+  def is_cargo_base_library(target):
+    return isinstance(target, CargoBaseLibrary)
+
+  @staticmethod
+  def is_cargo_base_binary(target):
+    return isinstance(target, CargoBaseBinary)
+
+  @staticmethod
+  def is_cargo_base_custom_build(target):
+    return isinstance(target, CargoBaseCustomBuild)
+
+  @staticmethod
+  def is_cargo_base_proc_macro(target):
+    return isinstance(target, CargoBaseProcMacro)
+
+  @staticmethod
+  def is_cargo_synthetic_library(target):
+    return isinstance(target, CargoSyntheticLibrary)
+
+  @staticmethod
+  def is_cargo_synthetic_binary(target):
+    return isinstance(target, CargoSyntheticBinary)
+
```

Comment:
My feeling is that the `is_X` methods might actually obscure things a little bit here. I would recommend inlining all of them, and waiting to reintroduce them until they are either used to represent a union of multiple parent classes... or, at that point, choosing to change the class hierarchy by introducing a mixin or other parent class instead.